contrib/99designs/gqlgen/tracer.go

```diff
@@ -0,0 +1,106 @@
+package gqlgen
+
+import (
+	"fmt"
+	"context"
+	"math"
+
+	"github.com/99designs/gqlgen/graphql"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+type trace struct {
+	analyticsRate float64
+}
+
+func New() graphql.Tracer {
+	return &trace{math.NaN()}
+}
+
+func (t *trace) StartOperationParsing(ctx context.Context) context.Context {
+//	_, ctx = tracer.StartSpanFromContext(ctx, "Parsing")
```

Comment:
It might be nice to have these parsing and validation spans if possible, but as far as I can tell, there's no easy way to integrate it that looks nice. There is no parent span, so the parsing and validation spans end up disjoint:

<img width="1008" alt="Screen Shot 2019-10-16 at 4 48 04 PM" src="https://user-images.githubusercontent.com/1819836/66961644-c1173500-f034-11e9-90e5-882766c2d65e.png">

I'm not sure how much value they add, though, so for now I think we can ignore them.

Reply:
I'm guessing from the code so far, the way this integration plugs into gqlgen is like: `handler := gqlgen.GraphQL(schema, gqlgen.Trace(gqlgentrace.New())`

If you want all those spans to have a parent, the handler (`http.HandlerFunc`) should create one.

I suspect the parsing and validating durations are useful in non-example code where it takes more than a few nanoseconds, and might grow exponentially for certain queries.